pkg/k8shandler/cluster.go

```diff
@@ -209,6 +220,22 @@ func getNodeUpgradeInProgress(cluster *api.Elasticsearch) NodeTypeInterface {
 	return nil
 }
 
+func progressUnshedulableNodes(cluster *api.Elasticsearch) {
+	for _, node := range cluster.Status.Nodes {
+		if _, condition := getESNodeCondition(node.Conditions, api.Unschedulable); condition != nil {
```

Comment:
since we don't do anything special with `condition` at all, can we make this a `hasCondition` instead and it return a `bool`?

Reply:
Yes, we can :)

For future reference, this code is my inspiration:
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/pod/util.go#L201
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/pod/util.go#L241-L256